### PR TITLE
domain: make TestSchemaValidator stable

### PR DESCRIPTION
**Pull request by zimulala**
### What problem does this PR solve? 
Make TestSchemaValidator stable.
```
FAIL: schema_validator_test.go:30: testSuite.TestSchemaValidator
schema_validator_test.go:66:
    c.Assert(valid, Equals, ResultUnknown)
... obtained domain.checkResult = 0
... expected domain.checkResult = 2
```

### What is changed and how it works?
Make sure that ts has timed out a lease and add comments.

Related changes

 - Need to cherry-pick to the release branch

---
This change is [Reviewable](https://reviewable.io/reviews/pingcap/tidb/8334)

### CODE REVIEW

File: domain/schema_validator_test.go [hunk 1]
```diff
@@ -63,10 +64,12 @@ func (*testSuite) TestSchemaValidator(c *C) {
 	validator.Restart()
 
 	// Sleep for a long time, check schema is invalid.
+	ts := <-oracleCh // Make sure that ts has timed out a lease.
```
Comment (shenli) [inline domain/schema_validator_test.go]: Should we move the comment to line 68?

Comment (zimulala) [reply domain/schema_validator_test.go]: No. It's used to comment line67-69.

**Comment by tiancaiamao** [comment]
LGTM

**Pull request merged**